code/modules/buildmode/submodes/delete.dm

```diff
@@ -0,0 +1,55 @@
+/datum/buildmode_mode/delete
+	key = "delete"
+
+/datum/buildmode_mode/delete/show_help(client/c)
+	to_chat(c, "<span class='notice'>***********************************************************</span>")
+	to_chat(c, "<span class='notice'>Left Mouse Button on anything to delete it. If you break it, you buy it.</span>")
+	to_chat(c, "<span class='notice'>Right Mouse Button on anything to delete everything of the type. Probably dont do this unless you know what you are doing.</span>")
+	to_chat(c, "<span class='notice'>***********************************************************</span>")
+
+/datum/buildmode_mode/delete/handle_click(client/c, params, obj/object)
+	var/list/pa = params2list(params)
+	var/left_click = pa.Find("left")
+	var/right_click = pa.Find("right")
+
+	if(left_click)
+		qdel(object)
+
+	if(right_click)
+		if(check_rights(R_DEBUG|R_SERVER))	//Prevents buildmoded non-admins from breaking everything.
+			var/action_type = alert("Strict type ([object.type]) or type and all subtypes?",,"Strict type","Type and subtypes","Cancel")
```

Comment:
Largely copied and adjusted for my needs from the similar VV dropdown option available on objects. Could probably be converted to a global proc and also expanded to all atoms in VV dropdown.